test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityTests.cs

```diff
@@ -42,14 +42,18 @@ public class CompatibilityTests
         [InlineData("net5.1", "netcoreapp3.1", true)]
         [InlineData("net6.0", "netcoreapp3.1", true)]
 
-        // net5.0-<osname> is not supported yet
-        [InlineData("net5.0-android", "net5.0", false)]
-        [InlineData("net5.0-android", "netcoreapp3.1", false)]
-        [InlineData("net5.0-ios", "net5.0", false)]
-        [InlineData("net5.0-macos", "net5.0", false)]
-        [InlineData("net5.0-tvos", "net5.0", false)]
-        [InlineData("net5.0-watchos", "net5.0", false)]
-        [InlineData("net5.0-windows", "net5.0", false)]
+        // net5.0-<osname> is compatible with net5.0 but not the inverse
+        [InlineData("net5.0-android", "net5.0", true)]
+        [InlineData("net5.0-android", "netcoreapp3.1", true)]
+        [InlineData("net5.0-ios", "net5.0", true)]
+        [InlineData("net5.0-macos", "net5.0", true)]
```

Comment:
>macos [](start = 28, length = 5)

do you have support and test coverage of Platform versions as well yet? coming later for this work? or in a separate task?